gluster-prometheus.spec

```diff
@@ -0,0 +1,89 @@
+%if 0%{?fedora}
+%global with_bundled 0
+%else
+%global with_bundled 1
+%endif
+
+%{!?with_debug: %global with_debug 1}
+
+%if 0%{?with_debug}
+%global _dwz_low_mem_die_limit 0
+%else
+%global debug_package   %{nil}
+%endif
+
+%{!?go_arches: %global go_arches x86_64 aarch64 ppc64le }
+
+%global provider github
+%global provider_tld com
+%global project gluster
+%global repo gluster-prometheus
+%global provider_prefix %{provider}.%{provider_tld}/%{project}/%{repo}
+%global import_path %{provider_prefix}
+%global gluster_mgmt %{getenv:GLUSTER_MGMT}
+
+%global gluster_prom_make %{__make} PREFIX=%{_prefix} EXEC_PREFIX=%{_exec_prefix} BINDIR=%{_bindir} SBINDIR=%{_sbindir} DATADIR=%{_datadir} LOCALSTATEDIR=%{_sharedstatedir} LOGDIR=%{_localstatedir}/log SYSCONFDIR=%{_sysconfdir} FASTBUILD=off
+
+%global gluster_prom_ver 1.0.0
+%global gluster_prom_rel rc0
+
+Name: %{repo}
+Version: %{gluster_prom_ver}
+Release: %{gluster_prom_rel}%{?dist}
+Summary: The GlusterFS prometheus metrics collectors
+License: GPLv2 or LGPLv3+
+URL: https://%{provider_prefix}
+%if 0%{?with_bundled}
+Source0: https://%{provider_prefix}/releases/download/v%{version}/gluster-exporter-v%{gluster_prom_ver}-%{gluster_prom_rel}-vendor.tar.xz
+%else
+Source0: https://%{provider_prefix}/releases/download/v%{version}/gluster-exporter-v%{gluster_prom_ver}-%{gluster_prom_rel}.tar.xz
+%endif
+
+ExclusiveArch: %{go_arches}
+
+BuildRequires: %{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
+BuildRequires: systemd
+
+%if ! 0%{?with_bundled}
+BuildRequires: golang(github-prometheus-client_golang)
```

Comment:
in fedora we have to mention BuildRequires like github.com/prometheus/client_golang/prometheus. Dont know about centos. otherwise rpmbuild complains about failed dependency.

Reply:
I feel its for bundled mode and should work on all the environments.
@aravindavk ^^^?

Reply:
we will bundle the dependencies only if not available in respective repository.

Refer: https://fedoraproject.org/wiki/PackagingDrafts/Go